### PR TITLE
Literal syntax

### PR DESCRIPTION
**Pull request by laurayuwen**
In this pull request, I finished my definition for lexical and literal syntax, and delete KLable definitions for klabels I put in attribute. 
Radu will work on this part for parsing purpose later so I skip when there is a problem (especially, when I find it hard to follow exactly IntLiteral, I made floatLiteral very simple).
Also, two small things: I seperated out JAVA-SYNTAX module in a seperate file and use it in both java-exec.k and java-prep.k. Denis used two JAVA-SYNTAX in each independently, but for me, they are the same.
I also sepeated out TYPE-SYNTAX module to be an independent file, which I will continue working on (and create a new branch TypeSyntax).

### CODE REVIEW

File: src/common/lexical-syntax.k [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                 | "throws"        
+                 | "transient"     
+                 | "try"           
+                 | "void"          
+                 | "volatile"      
+                 | "while"
+*/
+
+//@ \subsection{LineTerminators.sdf}
+syntax EndOfFile
```
Comment (andreistefanescu) [inline src/common/lexical-syntax.k]: What is the purpose of these 3 syntactic categories?

File: src/common/literal-syntax.k [hunk 2]
```diff
@@ -0,0 +1,50 @@
+require "lexical-syntax.k"
+require "list-syntax.k"
+module LITERAL-SYNTAX
+imports LEXICAL-SYNTAX
+imports LIST-SYNTAX
+//@ \subsection{BooleanLiterals.sdf}
+syntax BoolLiteral ::= Bool [klabel('Bool)]
```
Comment (andreistefanescu) [inline src/common/literal-syntax.k]: does this ```Bool``` conflict with the builtin one?

File: src/common/literal-syntax.k [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+syntax OctaEscape ::= Token{"\\" [0-7]}                  [onlyLabel, klabel('OctaEscape1)]
+                    | Token{"\\" [0-3][0-7]}             [onlyLabel, klabel('OctaEscape2)]
+                    | Token{"\\" [4-7][0-7]}             [onlyLabel, klabel('OctaEscape2)]
+                    | Token{"\\" [0-3][0-7][0-7]}        [onlyLabel, klabel('OctaEscape3)]
+
+//@ \subsection{FloatingPointLiterals.sdf} todo
+syntax FloatLiteral ::= Float      [klabel('Float)]
+
+//@ \subsection{IntegerLiterals.sdf}
+syntax DeciLiteral ::= Token{[1-9][0-9]* [lL]?} [onlyLabel]
```
Comment (andreistefanescu) [inline src/common/literal-syntax.k]: These are incomplete?

File: src/common/lexical-syntax.k (hunk 1 above)
Comment (laurayuwen) [inline src/common/lexical-syntax.k]: To denote line terminates or file ends. I just copied it from sdf, they are not needed?

File: src/common/literal-syntax.k (hunk 3 above)
Comment (laurayuwen) [inline src/common/literal-syntax.k]: For integers, sdf defines another sort, like DeciNumeral in between such that
 [1-9][0-9]*  -> DeciNumeral
 DeciNumeral [lL]? -> DeciLiteral

but adding this sort causes errors, and it is not further used, so I just define XXXLiteral

For Float, I just skipped those low level definitions, as long as I put klabel 'Float in the place I want.
